build/vsts-ci.yml

```diff
@@ -2,309 +2,317 @@
-      demands: ImageOverride -equals Build.Ubuntu.1804.Amd64
-    container: UbuntuCrossArm64Container
-    steps:
-    # Only build native assets to avoid conflicts.
-    - script: ./build.sh -configuration $(BuildConfig) -projects $(Build.SourcesDirectory)/src/Native/Native.proj /p:TargetArchitecture=arm64 /p:CopyPackageAssets=true
-      displayName: Build Native Assets
-
-    - task: PublishBuildArtifacts@1
-      displayName: Publish Linux_arm64 package assets
-      inputs:
-        pathToPublish: $(Build.SourcesDirectory)/artifacts/pkgassets
-        artifactName: pkgassets
-        artifactType: container
-
-    - template: /eng/common/templates/steps/generate-sbom.yml
-
-  ################################################################################
-  - job: MacOS
-  ################################################################################
-    pool:
-      vmImage: macOS-12
-    steps:
-    - script: export HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK=1 &&  rm '/usr/local/bin/2to3-3.11' && brew unlink libomp && brew install $(Build.SourcesDirectory)/build/libomp.rb --build-from-source --formula
-      displayName: Install build dependencies
-    # Only build native assets to avoid conflicts.
-    - script: ./build.sh -projects $(Build.SourcesDirectory)/src/Native/Native.proj -configuration $(BuildConfig) /p:TargetArchitecture=x64 /p:CopyPackageAssets=true
-      displayName: Build Native Assets
-
-    - task: PublishBuildArtifacts@1
-      displayName: Publish macOS package assets
-      inputs:
-        pathToPublish: $(Build.SourcesDirectory)/artifacts/pkgassets
-        artifactName: pkgassets
-        artifactType: container
-
-    - template: /eng/common/templates/steps/generate-sbom.yml
-
-  ################################################################################
-  - job: MacOS_Apple_Silicon
-  ################################################################################
-    pool:
-      vmImage: macOS-12
-    steps:
-    # Work around MacOS Homebrew image/environment bug: https://github.com/actions/virtual-environments/issues/2322#issuecomment-749211076
-    - script: |
-        rm -rf /usr/local/bin/2to3
-      displayName: MacOS Homebrew bug Workaround
-      continueOnError: true
-    - script: brew update && brew install libomp && brew link libomp --force
-      displayName: Install build dependencies
-    # Only build native assets to avoid conflicts.
-    - script: ./build.sh -projects $(Build.SourcesDirectory)/src/Native/Native.proj -configuration $(BuildConfig) /p:TargetArchitecture=arm64 /p:CopyPackageAssets=true
-      displayName: Build Native Assets
-
-    - task: PublishBuildArtifacts@1
-      displayName: Publish macOS_M1 package assets
-      inputs:
-        pathToPublish: $(Build.SourcesDirectory)/artifacts/pkgassets
-        artifactName: pkgassets
-        artifactType: container
-
-    - template: /eng/common/templates/steps/generate-sbom.yml
+- name: BuildConfig
```

Comment:
Do we have to use this multi-line `name:`  `value:` syntax, or can we use single-line `<name>: <value>` as was done before?

Reply:
According to the documentation, you can't mix both `<name>: <value>` syntax with use of `template:`, and switching to the more ugly `name: <name>\nvalue:<value>` syntax cleared those problems. Alternatively you can use multiple `variables:` blocks, but I believe that causes the variables to be namespaced, and wanted to avoid that

Reply:
Crud, I was never successful at getting it to work myself.  I was hoping you knew the trick.